plugins/chat/lib/chat/message_processor.rb

```diff
@@ -22,7 +22,12 @@ def run!
     end
 
     def process_thumbnails
+      return if !SiteSetting.create_thumbnails
+      filetypes = %w[jpg jpeg png heic]
+
       @model.uploads.each do |upload|
+        next if !upload.present? || !filetypes.include?(upload.extension.downcase)
```

Comment:
Could avoid negation on both conditions here:

```ruby
next if upload.blank? || filetypes.exclude?(upload.extension.downcase)
```

Or if we're okay with safe navigation:

```ruby
next if filetypes.exclude?(upload&.extension&.downcase)
```